src/Console/Output/Progress/DotsOutput.php

```diff
@@ -125,9 +113,14 @@ public function printLegend(): void
                 continue;
             }
 
-            $symbols[$symbol] = sprintf('%s-%s', $this->output->isDecorated() ? sprintf($status['format'], $symbol) : $symbol, $status['description']);
+            $symbols[$symbol] = sprintf('%s-%s', $this->output()->isDecorated() ? sprintf($status['format'], $symbol) : $symbol, $status['description']);
         }
 
-        $this->output->write(sprintf("\nLegend: %s\n", implode(', ', $symbols)));
+        $this->output()->write(sprintf("\nLegend: %s\n", implode(', ', $symbols)));
+    }
+
+    private function output(): OutputInterface
```

Comment:
tbh I don't see value of this method - it's typical proxy method.
yes, i see that one can use this method and have few chars shorter code - but also ppl will be able to use 2 variants in this class to get `OutputInterface` - `$this->output()` or `$this->context->getOutput()`.

btw, I don't get why sometimes
```php
function getOutput(): OutputInterface
```
and sometimes
```php
function output(): OutputInterface
```

Reply:
Yes, it's only shortcut to not repeat `$this->context->getOutput()`. Personally I don't like getters with `get*` prefix, it's redundant for me. Method's name itself should be descriptive enough to tell developer if the intention is to access something (get) or make some action (set).

Reply:
pls avoid proxy methods, there should be one way to get given value, otherwise those 2 "for now identically resulted" approaches will diverge over time - and till they will, it will still cause confusion which variant to use.

for whole project, we use `getX` approach.

Reply:
This is private method and is used consistently within whole class (6 times), there's no long form usage. I don't agree about diverging and confusion, and I believe `$this->output()->write()` is a lot more readable and meaningful than `$this->context->getOutput()->write()` 🤷‍♂️. I can change it if you really want, but I would prefer to keep it as-is.

PS. For public APIs I keep `getX()` convention (as you can see in `src/Console/Output/OutputContext.php`), but here I wanted to make code less cluttered.

Reply:
let's use `getX` for whole project, as we do right now. including `OutputContext`. Methods should have verbs, properties should not.